src/game/GameState.ts

```diff
@@ -0,0 +1,76 @@
+import { Field, Struct, MerkleMap, PublicKey, UInt32 } from 'snarkyjs';
+
+import { Piece } from '../objects/Piece';
+import { Position } from '../objects/Position';
+import { Unit } from '../objects/Unit';
+
+export class GameState extends Struct({
+  piecesRoot: Field, // root hash of pieces in the arena keyed by their id
+  arenaRoot: Field, // root hash of a merkle map of positions which are occupied
+  playerTurn: Field,
+  player1: PublicKey,
+  player2: PublicKey,
+  arenaLength: UInt32,
+  arenaWidth: UInt32,
+  turnsCompleted: UInt32,
+}) {
+  static empty(player1: PublicKey, player2: PublicKey): GameState {
+    const pieces = new MerkleMap();
+    const arena = new MerkleMap();
+
+    const pos1 = Position.fromXY(0, 0);
+    const pos2 = Position.fromXY(0, 15);
+    const pos3 = Position.fromXY(15, 0);
+    const pos4 = Position.fromXY(15, 15);
```

Comment:
Not an issue for now, but when we eventually implement player executed unit deployment, do we imagine that would be fed into this `empty` method, or would `empty` be called first and then the deployment would be sort of a one-time Phase of the game?

Reply:
I am moving away from the `empty` function in general.  It's definitely not a part of the proof.

What I imagine is that the server will deploy all the pieces however it wants and the proof will just start from there.  Currently there are no rules on starting a game, but I'll come back to that.